src/config_consumer.rs

```diff
@@ -0,0 +1,232 @@
+        let example = vec![
+            0x84, 0xaf, 0x73, 0x75, 0x62, 0x73, 0x63, 0x72, 0x69, 0x70, 0x74, 0x69, 0x6f, 0x6e,
+            0x5f, 0x69, 0x64, 0xd9, 0x20, 0x64, 0x37, 0x36, 0x32, 0x39, 0x63, 0x36, 0x63, 0x38,
+            0x32, 0x62, 0x65, 0x34, 0x66, 0x36, 0x37, 0x39, 0x65, 0x65, 0x37, 0x38, 0x61, 0x30,
+            0x64, 0x39, 0x37, 0x37, 0x38, 0x35, 0x36, 0x64, 0x32, 0xa3, 0x75, 0x72, 0x6c, 0xb0,
+            0x68, 0x74, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x73, 0x65, 0x6e, 0x74, 0x72, 0x79, 0x2e,
+            0x69, 0x6f, 0xb0, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x76, 0x61, 0x6c, 0x5f, 0x73, 0x65,
+            0x63, 0x6f, 0x6e, 0x64, 0x73, 0xcd, 0x01, 0x2c, 0xaa, 0x74, 0x69, 0x6d, 0x65, 0x6f,
+            0x75, 0x74, 0x5f, 0x6d, 0x73, 0xcd, 0x01, 0xf4,
+        ];
+
+        let example_uuid = uuid!("d7629c6c-82be-4f67-9ee7-8a0d977856d2");
+
+        let message = Message {
+            inner_message: InnerMessage::BrokerMessage(BrokerMessage {
+                partition: Partition {
+                    index: 2,
+                    topic: Topic::new("uptime-configs"),
+                },
+                payload: KafkaPayload::new(Some(example_uuid.into()), None, Some(example)),
+                offset: 0,
+                timestamp: Utc::now(),
+            }),
+        };
+        let _ = update_config_store(config_store.clone(), message);
+
+        let configs = config_store.read().unwrap().all_configs();
+
+        assert_eq!(configs.len(), 1);
+        assert_eq!(configs[0].partition, 2);
+        assert_eq!(
+            configs[0].subscription_id,
+            uuid!("d7629c6c-82be-4f67-9ee7-8a0d977856d2")
+        );
+    }
+
+    #[test]
+    fn test_drop_config() {
+        let config_store = Arc::new(ConfigStore::new_rw());
+
+        let example_config = Arc::new(CheckConfig::default());
+
+        config_store
+            .write()
+            .unwrap()
+            .add_config(example_config.clone());
+
+        // Empty message which will be ready for log compaction
+        let message = Message {
+            inner_message: InnerMessage::BrokerMessage(BrokerMessage {
+                partition: Partition {
+                    index: 2,
+                    topic: Topic::new("uptime-configs"),
+                },
+                payload: KafkaPayload::new(
+                    Some(example_config.clone().subscription_id.into()),
+                    None,
+                    None,
+                ),
+                offset: 0,
+                timestamp: Utc::now(),
+            }),
+        };
+        let _ = update_config_store(config_store.clone(), message);
+
+        let configs = config_store.read().unwrap().all_configs();
+
+        // example_config was removed
+        assert_eq!(configs.len(), 0);
+        assert_eq!(Arc::strong_count(&example_config), 1);
+    }
+
+    // TODO(epurkhiser): We probably want to test the update_partition callback
```

Comment:
I think i need to actually run kafka to test this?

Reply:
Arroyo contains types for producing/consuming "locally", i.e. without running Kafka (`LocalConsumer`, `LocalBroker`, &c.). You might be able to make it work with those.

Reply:
cool, I'll try and dig into this.

Reply:
Will follow up on this later. Going to merge for now.